### DISCUSSION TITLE
I want to re-name a directory. How do I redirect so that my existing links aren't broken?

### DISCUSSION THREAD

**Discussion by infominer33** [community/community#22301] [Other Feature Feedback, Questions, & Ideas]
<p>Trying to search google for this has me frustrated. can anyone help re-structuring site directory without breaking existing links?</p>

**Comment by oycyc** [comment]
<p>You can do this via two ways, by using the terminal or the GitHub web interface.</p>
<p>Using the terminal:</p>
<p>1.) Clone your repository that you website is hosted on.</p>
<p>2.) cd “repositoryname”</p>
<p>3.) git mv “directoryname” “newdirectoryname”</p>
<p>4.) git add .</p>
<p>5.) git commit -m “message”</p>
<p>6.) git push</p>
<p>This is the easiest way, by using the terminal. <a href="https://youtu.be/1LUEMI117TE" rel="nofollow noopener">Click here for a video of me doing this using the terminal.</a></p>
<p>However, if you want to do it using the web interface, you can achieve the same results, but it requires more time and it is very tedious if there is a lot of files in that directory. You just have to open every single file in that directory, go to the first letter of the file name and backspace. This should remove that “/” from the path, and you will have to rename every single file’s directory like that. <a href="https://youtu.be/X15GpFFcz8M" rel="nofollow noopener">Here is a video if you need help.</a></p>

**Comment by infominer33** [comment, marked as answer]
<p>sorry, I didn’t explain myself completely.</p>
<p>What I mean, is I want to change the name of a folder within the repository and I want the links that breaks to be re-directed to the new location of the resource.</p>
<p>For example, I want infominer.id/podcast-transcripts/index.html</p>
<p>to redirect to the new location infominer.id/transcripts/index.html.</p>
<p>[edit]</p>
<p>Looks like I’ve found my answer, finally:</p>
<p><a href="https://help.github.com/en/articles/redirects-on-github-pages" rel="nofollow noopener">https://help.github.com/en/articles/redirects-on-github-pages</a></p>
